src/components/animal/AnimalDetail.vue

```diff
@@ -1,65 +1,108 @@
+          <AnimalDatum :fieldData="getFieldData('race')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('description')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('physicalDescription')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('attitudeDescription')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('likes')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('dislikes')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('vaccines')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('nutrition')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('origin')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('chip')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('registerDate')" @update-field="updateAnimal"/>
+          <AnimalDatum :fieldData="getFieldData('identifier')" @update-field="updateAnimal" />
+
         </div>
+        <div class="column is-4">
+          <figure class="image">
+            <img src="https://dummyimage.com/320x200/000/fff"/>
+          </figure>
+          <button @click="deleteAnimal" class="button is-danger">Supprimer</button>
+        </div>
+      </div>
     </div>
+  </div>
 </template>
 <script>
 import AnimalDatum from './AnimalDatum.vue'
+
 export default {
-    components: {
-        AnimalDatum
+  components: {
+    AnimalDatum
+  },
+  props: ['id'],
+  data() {
+    return {
+      data: {},
+      extended: false,
+      loading: true,
+    }
+  },
+  async mounted() {
+    await this.loadAnimal();
+  },
+  methods: {
+    getFieldData(key){
+      return {key: key, value: this.data[key], editable: false}
+    },
+    async loadAnimal() {
+      this.loading = true;
+      try {
+        const res = await fetch(process.env.VUE_APP_BASE_URL + '/animals/' + this.id, {
+          method: 'get',
+          headers: {
+            'content-type': 'application/json'
+          }
+        })
+        this.data = await res.json()
+      } catch (err) {
+        console.log(err);
+      } finally {
+        this.loading = false;
+      }
+    },
+    async deleteAnimal() {
+      try {
+        await fetch(process.env.VUE_APP_BASE_URL + '/animals/' + this.data.id, {
+          method: 'DELETE',
+          headers: {
+            'content-type': 'application/json'
+          }
+        });
+      } catch (err) {
+        console.log(err);
+      } finally {
+        this.$emit('delete-animal', this.data.id)
+      }
     },
-    props: ['data'],
-    data() {
-        return {
-            extended: false
-        }
+    updateAnimal(newData) {
+      for (let i in newData) {
+        this.data[i] = newData[i];
+      }
+      this.saveAnimal();
```

Comment:
```suggestion
     
```
Ajouter peut-être l'apparition de bouton "Valider modification" (qui appelle `this.saveAnimal`) - "Annuler modification" (qui remet l'ensemble des champs comme avant) une fois qu'un champ est modifié, ce qui justifierait l'utilisation d'un event pour la vue `AnimalDatum`.

Reply:
C'est exactement ce que je voulais faire et je pense que l'on va faire ceci

Reply:
Ajouté au dernier commit